src/main/java/vazkii/quark/content/world/block/AncientSaplingBlock.java

```diff
@@ -48,89 +53,144 @@ public static class AncientTree extends AbstractTreeGrower {
-			this.folliageDistance = folliageDistance;
+			this.foliageDistance = foliageDistance;
 			this.maxBlobs = maxBlobs;
+			this.freeTopHeight = freeTopHeight;
 		}
 
 		@Override
-		public List<FoliagePlacer.FoliageAttachment> placeTrunk(LevelSimulatedReader p_226147_, BiConsumer<BlockPos, BlockState> p_226148_, RandomSource p_226149_, int p_226150_, BlockPos p_226151_, TreeConfiguration p_226152_) {
-			setDirtAt(p_226147_, p_226148_, p_226149_, p_226151_.below(), p_226152_);
+		public List<FoliagePlacer.FoliageAttachment> placeTrunk(LevelSimulatedReader level, BiConsumer<BlockPos, BlockState> blockSetter, RandomSource random, int freeTreeHeight, BlockPos pos, TreeConfiguration config) {
+			setDirtAt(level, blockSetter, random, pos.below(), config);
 
-			List<BlockPos> folliagePositions = new ArrayList<>();
-			
+			List<BlockPos> foliagePositions = new ArrayList<>();
+
+			int sampledMaxBlobs = maxBlobs.sample(random);
 			int placed = 0;
-			int j = 0;
-			for(int i = p_226150_; i >= 0; --i) {
-				BlockPos target = p_226151_.above(i);
-				this.placeLog(p_226147_, p_226148_, p_226149_, target, p_226152_);
+			int j = freeTopHeight.sample(random);
+			for(int i = freeTreeHeight; i >= 0; --i) {
+				BlockPos target = pos.above(i);
+				this.placeLog(level, blockSetter, random, target, config);
 				
-				if(placed < maxBlobs) {
+				if(placed < sampledMaxBlobs) {
 					if(j == 0) {
-						folliagePositions.add(target);
-						j = folliageDistance;
+						foliagePositions.add(target);
+						j = foliageDistance.sample(random);
 						placed++;
 					} else j--;
 				}
 				
 			}
 
-			return folliagePositions.stream().map(p -> new FoliagePlacer.FoliageAttachment(p, 0, false)).collect(Collectors.toList());
+			return foliagePositions.stream().map(p -> new FoliagePlacer.FoliageAttachment(p, 0, false)).collect(Collectors.toList());
 		}
 
 		@Override
 		protected TrunkPlacerType<?> type() {
-			return null;
+			return Type.ANCIENT_TREE_TOPPER;
+		}
+
+		public static class Type extends TrunkPlacerType<MultiFoliageStraightTrunkPlacer> {
+			public static final TrunkPlacerType<MultiFoliageStraightTrunkPlacer> ANCIENT_TREE_TOPPER = register();
+
+			private static final Codec<MultiFoliageStraightTrunkPlacer> CODEC = RecordCodecBuilder.create(
+				instance -> trunkPlacerParts(instance)
+					.and(IntProvider.NON_NEGATIVE_CODEC.fieldOf("foliage_distance").forGetter(trunkPlacer -> trunkPlacer.foliageDistance))
+					.and(IntProvider.NON_NEGATIVE_CODEC.fieldOf("max_blobs").forGetter(trunkPlacer -> trunkPlacer.maxBlobs))
+					.and(IntProvider.NON_NEGATIVE_CODEC.fieldOf("free_top_height").forGetter(trunkPlacer -> trunkPlacer.freeTopHeight))
+					.apply(instance, MultiFoliageStraightTrunkPlacer::new)
+			);
+
+			private static TrunkPlacerType<MultiFoliageStraightTrunkPlacer> register() {
+				Type t = new Type();
+				RegistryHelper.register(t, "multi_foliage_straight_trunk_placer", Registry.TRUNK_PLACER_TYPE_REGISTRY);
```

Comment:
The problem is here. Despite this being called, it doesn't actually end up in the registry. At least not with a name i expect.